topi/include/topi/transform.h

```diff
@@ -358,5 +358,66 @@ inline Array<Tensor> split_sections(const Tensor& x,
   return split(x, split_indices, axis, name, tag);
 }
 
+/*!
+* \brief Take elements from an array along an axis.
+*
+* \param a The source array.
+* \param indices The indices of the values to extract.
+* \param axis The axis over which to select values. By default,
+* the flattened input array is used.
+* \param name The name of the operation.
+* \param tag The tag to mark the operation.
+*
+* \return A Tensor whose op member is the take operation
+*/
+inline Tensor take(const Tensor& a,
+                  const Tensor& indices,
+                  const int* axis = NULL,
+                  std::string name = "tensor",
+                  std::string tag = kInjective) {
+  Tensor a_ary = a;
+  if (axis == NULL) {
+    int a_size = 1;
+    for (size_t posi = 0; posi < a_ary->shape.size(); ++posi) {
+      a_size = a_size * static_cast<int>(GetConstInt(a_ary->shape[posi]));
+    }
+    Array<Expr> a_shape_exp;
+    a_shape_exp.push_back(a_size);
+    a_ary = reshape(a_ary, a_shape_exp);
```

Comment:
again, do not use reshape operator, try to inline the compute expression everything and directly write a take_flatten function